util/process_wrapper/BUILD.bazel

```diff
@@ -1,26 +1,40 @@
-load("@rules_cc//cc:defs.bzl", "cc_binary")
+load("//rust:defs.bzl", "rust_binary", "rust_test")
 
-cc_binary(
+# buildifier: disable=bzl-visibility
+load("//rust/private:transitions.bzl", "without_process_wrapper")
+
+alias(
     name = "process_wrapper",
-    srcs = [
-        "process_wrapper.cc",
-        "system.h",
-        "utils.h",
-        "utils.cc",
-    ] + select({
-        "@bazel_tools//src/conditions:host_windows": [
-            "system_windows.cc",
-        ],
-        "//conditions:default": [
-            "system_posix.cc",
-        ],
-    }),
-    defines = [] + select({
-        "@bazel_tools//src/conditions:host_windows": [
-            "UNICODE",
-            "_UNICODE",
-        ],
-        "//conditions:default": [],
+    actual = select({
+        # This will never get used, it's only here to break the circular dependency to allow building process_wrapper
+        ":use_fake_process_wrapper": ":process_wrapper_fake",
```

Comment:
Am I blind? Where is `process_wrapper_fake` being defined?

Reply:
What a nerd-snipe @gigaroby :) You are not blind, this is a funny tricky edge case in Bazel. I'm not going to admit how much time I've just spent trying to understand this :) `process_wrapper_fake` is only accessed through `alias`. Bazel checks if the target of the alias is a generated file, and if it is not, Bazel assumes it's a source file. Bazel actually doesn't check if this source file exists and happily goes ahead, probably assuming the build will fail in execution. Since we don't use the file in execution, there is nothing to fail. Beautiful.

@gigaroby could you please add an empty target for this anyway? As it stands now this is super confusing.

@comius just in case you were not aware, or this is actually a bug (I think this may have been intentional "optimization" but I can be wrong). Bazel seems to be not checking if a target of an alias exists.

Reply:
Done.